{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import pandas as pd\n",
    "import os\n",
    "import itertools\n",
    "import numpy as np\n",
    "from sklearn import preprocessing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set up the working directory\n",
    "# All paths after this are relative path to the root of chead\n",
    "\n",
    "wk_dir = '/Users/hxia/Desktop/BBL/' # this is where I have mounted chead to on my laptop. \n",
    "# Change the above to an empty string if working directly on chead \n",
    "\n",
    "# Set up the project directory\n",
    "project_path = os.path.join(wk_dir,'data/joy/BBL/projects/prsConnectivity')\n",
    "\n",
    "pd.options.display.max_rows = 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the list of Genes and their associated GWAS p-val from Ripke et al. Nature 2014\n",
    "# Article and its supplement from www.nature.com/articles/nature13595\n",
    "\n",
    "gene_nature_supplement_file = os.path.join(project_path,'data','nature108genes.csv')\n",
    "gene_nature_supplement = pd.read_csv(gene_nature_supplement_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Match aliases of genes\n",
    "scz_gene_list  = list(gene_nature_supplement['Gene1'])\n",
    "scz_gene_list = ['LBA1' if gene=='TRANK1' else gene for gene in scz_gene_list]\n",
    "scz_gene_list = ['RP11-259P6.1' if gene=='IGSF9B' else gene for gene in scz_gene_list]\n",
    "scz_gene_list = ['AP001931.1' if gene=='BTBD18' else gene for gene in scz_gene_list]\n",
    "scz_gene_list = ['RGS6' if gene=='AC005477.1' else gene for gene in scz_gene_list]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the list of genes sampled by Allen Brain Institute\n",
    "donor = 12876\n",
    "donor_gene_path = os.path.join(project_path,'ABI','processedData','donor%s_gene_expression.csv' % donor)\n",
    "donor_gene = pd.read_csv(donor_gene_path)\n",
    "\n",
    "# Get the list of tissue locations\n",
    "gene_location = pd.read_csv(os.path.join(project_path,'ABI','normalized_microarray_donor%s' % donor,'SampleAnnot.csv'))\n",
    "\n",
    "# Get the list of genes-mapping-to-Schaefer\n",
    "numParcels = 100\n",
    "numNetworks = 7\n",
    "mm = 1\n",
    "gene_Scachefer_path = os.path.join(project_path,'ABI','gene_mapping','%sdonor_%sParcels_%sNetwork_%smm.csv' \\\n",
    "                                % (donor,numParcels,numNetworks,mm))\n",
    "donor_gene_parcel = pd.read_csv(gene_Scachefer_path)\n",
    "\n",
    "# Combine the above 2 lists\n",
    "donor_gene_parcel_location = pd.concat([gene_location,donor_gene_parcel[['parcel','community']]],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Should we standardize or or normalize the data?\n",
    "#donor_gene.iloc[:,2:] = preprocessing.scale(donor_gene.iloc[:,2:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>353</th>\n",
       "      <th>354</th>\n",
       "      <th>355</th>\n",
       "      <th>356</th>\n",
       "      <th>357</th>\n",
       "      <th>358</th>\n",
       "      <th>359</th>\n",
       "      <th>360</th>\n",
       "      <th>361</th>\n",
       "      <th>362</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>...</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "      <td>2.073800e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>-1.548680e-16</td>\n",
       "      <td>2.508040e-16</td>\n",
       "      <td>-2.055770e-17</td>\n",
       "      <td>-1.014180e-16</td>\n",
       "      <td>2.069475e-16</td>\n",
       "      <td>1.726847e-16</td>\n",
       "      <td>-9.593595e-18</td>\n",
       "      <td>2.549155e-16</td>\n",
       "      <td>-1.562385e-16</td>\n",
       "      <td>-1.110116e-16</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.370514e-18</td>\n",
       "      <td>2.329873e-16</td>\n",
       "      <td>2.055770e-16</td>\n",
       "      <td>-1.233462e-17</td>\n",
       "      <td>-1.041590e-16</td>\n",
       "      <td>1.795373e-16</td>\n",
       "      <td>-3.056245e-16</td>\n",
       "      <td>7.126670e-17</td>\n",
       "      <td>-7.948978e-17</td>\n",
       "      <td>-4.317118e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "      <td>1.000024e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>-1.644064e+00</td>\n",
       "      <td>-1.694469e+00</td>\n",
       "      <td>-1.691303e+00</td>\n",
       "      <td>-1.686374e+00</td>\n",
       "      <td>-1.622112e+00</td>\n",
       "      <td>-1.680970e+00</td>\n",
       "      <td>-1.634586e+00</td>\n",
       "      <td>-1.660567e+00</td>\n",
       "      <td>-1.689539e+00</td>\n",
       "      <td>-1.701901e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.666306e+00</td>\n",
       "      <td>-1.660875e+00</td>\n",
       "      <td>-1.653132e+00</td>\n",
       "      <td>-1.683748e+00</td>\n",
       "      <td>-1.646011e+00</td>\n",
       "      <td>-1.629923e+00</td>\n",
       "      <td>-1.657872e+00</td>\n",
       "      <td>-1.682752e+00</td>\n",
       "      <td>-1.669342e+00</td>\n",
       "      <td>-1.651918e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>-8.222022e-01</td>\n",
       "      <td>-8.191775e-01</td>\n",
       "      <td>-8.420820e-01</td>\n",
       "      <td>-8.564225e-01</td>\n",
       "      <td>-8.649453e-01</td>\n",
       "      <td>-8.326618e-01</td>\n",
       "      <td>-8.350242e-01</td>\n",
       "      <td>-8.544921e-01</td>\n",
       "      <td>-8.424799e-01</td>\n",
       "      <td>-7.881720e-01</td>\n",
       "      <td>...</td>\n",
       "      <td>-8.269119e-01</td>\n",
       "      <td>-8.613034e-01</td>\n",
       "      <td>-8.787918e-01</td>\n",
       "      <td>-8.159170e-01</td>\n",
       "      <td>-8.584238e-01</td>\n",
       "      <td>-8.628142e-01</td>\n",
       "      <td>-8.332668e-01</td>\n",
       "      <td>-7.951214e-01</td>\n",
       "      <td>-8.242145e-01</td>\n",
       "      <td>-8.315368e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>5.387942e-02</td>\n",
       "      <td>5.383682e-02</td>\n",
       "      <td>4.122789e-02</td>\n",
       "      <td>5.312408e-02</td>\n",
       "      <td>2.896458e-02</td>\n",
       "      <td>5.303130e-02</td>\n",
       "      <td>5.212051e-02</td>\n",
       "      <td>4.470100e-02</td>\n",
       "      <td>4.789715e-02</td>\n",
       "      <td>4.571557e-02</td>\n",
       "      <td>...</td>\n",
       "      <td>5.081153e-02</td>\n",
       "      <td>3.964970e-02</td>\n",
       "      <td>3.036516e-02</td>\n",
       "      <td>4.333802e-02</td>\n",
       "      <td>6.735844e-02</td>\n",
       "      <td>4.481208e-02</td>\n",
       "      <td>6.833994e-02</td>\n",
       "      <td>5.445278e-02</td>\n",
       "      <td>5.008767e-02</td>\n",
       "      <td>2.721728e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>7.261703e-01</td>\n",
       "      <td>7.268089e-01</td>\n",
       "      <td>7.265178e-01</td>\n",
       "      <td>7.349928e-01</td>\n",
       "      <td>7.253707e-01</td>\n",
       "      <td>7.299048e-01</td>\n",
       "      <td>7.259513e-01</td>\n",
       "      <td>7.261852e-01</td>\n",
       "      <td>7.208755e-01</td>\n",
       "      <td>7.116153e-01</td>\n",
       "      <td>...</td>\n",
       "      <td>7.248224e-01</td>\n",
       "      <td>7.249376e-01</td>\n",
       "      <td>7.245606e-01</td>\n",
       "      <td>7.194416e-01</td>\n",
       "      <td>7.311956e-01</td>\n",
       "      <td>7.324349e-01</td>\n",
       "      <td>7.325194e-01</td>\n",
       "      <td>7.139762e-01</td>\n",
       "      <td>7.204163e-01</td>\n",
       "      <td>7.227240e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>3.556491e+00</td>\n",
       "      <td>3.735611e+00</td>\n",
       "      <td>3.744245e+00</td>\n",
       "      <td>3.644730e+00</td>\n",
       "      <td>3.729151e+00</td>\n",
       "      <td>3.668263e+00</td>\n",
       "      <td>3.541770e+00</td>\n",
       "      <td>3.768874e+00</td>\n",
       "      <td>3.743219e+00</td>\n",
       "      <td>3.700943e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>3.655365e+00</td>\n",
       "      <td>3.672944e+00</td>\n",
       "      <td>3.641652e+00</td>\n",
       "      <td>3.666557e+00</td>\n",
       "      <td>3.782848e+00</td>\n",
       "      <td>3.661159e+00</td>\n",
       "      <td>3.465495e+00</td>\n",
       "      <td>3.493549e+00</td>\n",
       "      <td>3.674348e+00</td>\n",
       "      <td>3.470825e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 363 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                0             1             2             3             4    \\\n",
       "count  2.073800e+04  2.073800e+04  2.073800e+04  2.073800e+04  2.073800e+04   \n",
       "mean  -1.548680e-16  2.508040e-16 -2.055770e-17 -1.014180e-16  2.069475e-16   \n",
       "std    1.000024e+00  1.000024e+00  1.000024e+00  1.000024e+00  1.000024e+00   \n",
       "min   -1.644064e+00 -1.694469e+00 -1.691303e+00 -1.686374e+00 -1.622112e+00   \n",
       "25%   -8.222022e-01 -8.191775e-01 -8.420820e-01 -8.564225e-01 -8.649453e-01   \n",
       "50%    5.387942e-02  5.383682e-02  4.122789e-02  5.312408e-02  2.896458e-02   \n",
       "75%    7.261703e-01  7.268089e-01  7.265178e-01  7.349928e-01  7.253707e-01   \n",
       "max    3.556491e+00  3.735611e+00  3.744245e+00  3.644730e+00  3.729151e+00   \n",
       "\n",
       "                5             6             7             8             9    \\\n",
       "count  2.073800e+04  2.073800e+04  2.073800e+04  2.073800e+04  2.073800e+04   \n",
       "mean   1.726847e-16 -9.593595e-18  2.549155e-16 -1.562385e-16 -1.110116e-16   \n",
       "std    1.000024e+00  1.000024e+00  1.000024e+00  1.000024e+00  1.000024e+00   \n",
       "min   -1.680970e+00 -1.634586e+00 -1.660567e+00 -1.689539e+00 -1.701901e+00   \n",
       "25%   -8.326618e-01 -8.350242e-01 -8.544921e-01 -8.424799e-01 -7.881720e-01   \n",
       "50%    5.303130e-02  5.212051e-02  4.470100e-02  4.789715e-02  4.571557e-02   \n",
       "75%    7.299048e-01  7.259513e-01  7.261852e-01  7.208755e-01  7.116153e-01   \n",
       "max    3.668263e+00  3.541770e+00  3.768874e+00  3.743219e+00  3.700943e+00   \n",
       "\n",
       "           ...                353           354           355           356  \\\n",
       "count      ...       2.073800e+04  2.073800e+04  2.073800e+04  2.073800e+04   \n",
       "mean       ...      -1.370514e-18  2.329873e-16  2.055770e-16 -1.233462e-17   \n",
       "std        ...       1.000024e+00  1.000024e+00  1.000024e+00  1.000024e+00   \n",
       "min        ...      -1.666306e+00 -1.660875e+00 -1.653132e+00 -1.683748e+00   \n",
       "25%        ...      -8.269119e-01 -8.613034e-01 -8.787918e-01 -8.159170e-01   \n",
       "50%        ...       5.081153e-02  3.964970e-02  3.036516e-02  4.333802e-02   \n",
       "75%        ...       7.248224e-01  7.249376e-01  7.245606e-01  7.194416e-01   \n",
       "max        ...       3.655365e+00  3.672944e+00  3.641652e+00  3.666557e+00   \n",
       "\n",
       "                357           358           359           360           361  \\\n",
       "count  2.073800e+04  2.073800e+04  2.073800e+04  2.073800e+04  2.073800e+04   \n",
       "mean  -1.041590e-16  1.795373e-16 -3.056245e-16  7.126670e-17 -7.948978e-17   \n",
       "std    1.000024e+00  1.000024e+00  1.000024e+00  1.000024e+00  1.000024e+00   \n",
       "min   -1.646011e+00 -1.629923e+00 -1.657872e+00 -1.682752e+00 -1.669342e+00   \n",
       "25%   -8.584238e-01 -8.628142e-01 -8.332668e-01 -7.951214e-01 -8.242145e-01   \n",
       "50%    6.735844e-02  4.481208e-02  6.833994e-02  5.445278e-02  5.008767e-02   \n",
       "75%    7.311956e-01  7.324349e-01  7.325194e-01  7.139762e-01  7.204163e-01   \n",
       "max    3.782848e+00  3.661159e+00  3.465495e+00  3.493549e+00  3.674348e+00   \n",
       "\n",
       "                362  \n",
       "count  2.073800e+04  \n",
       "mean  -4.317118e-17  \n",
       "std    1.000024e+00  \n",
       "min   -1.651918e+00  \n",
       "25%   -8.315368e-01  \n",
       "50%    2.721728e-02  \n",
       "75%    7.227240e-01  \n",
       "max    3.470825e+00  \n",
       "\n",
       "[8 rows x 363 columns]"
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(preprocessing.scale(donor_gene.iloc[:,2:])).describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Allen Brain sampled 96 genes of the 108 SCZ loci\n"
     ]
    }
   ],
   "source": [
    "# Calcualte the overlap between SCZ genes and ABI genes\n",
    "gene_overlap = []\n",
    "gene_non_overlap = []\n",
    "for scz_gene in scz_gene_list:\n",
    "    if scz_gene in list(donor_gene['gene_symbol']):\n",
    "        gene_overlap.append(scz_gene)\n",
    "    else: gene_non_overlap.append(scz_gene) \n",
    "        \n",
    "print('Allen Brain sampled %s genes of the 108 SCZ loci' % len(gene_overlap))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Assemble a dataframe\n",
    "gene_overlap_exp = pd.DataFrame()\n",
    "gene_p_val_dict = []\n",
    "gene_p_val_list = []\n",
    "\n",
    "# Loop through each overlap gene and assign them to a new DF\n",
    "# Loop through each overlap gene and get the GWAS p-val\n",
    "for gene in gene_overlap:\n",
    "    gene_call = gene\n",
    "    if gene == 'LBA1': gene_call = 'TRANK1' \n",
    "    if gene == 'RP11-259P6.1': gene_call = 'IGSF9B'\n",
    "    if gene == 'AP001931.1': gene_call = 'BTBD18'\n",
    "    if gene == 'RGS6': gene_call = 'AC005477.1'\n",
    "    gene_overlap_exp = pd.concat([gene_overlap_exp,donor_gene[donor_gene['gene_symbol']==gene]])\n",
    "    gene_p_val = [gene_nature_supplement['P-value'][gene_nature_supplement['Gene1'] == gene_call].get_values()][0][0]\n",
    "    gene_p_val_dict.append({gene: gene_p_val})\n",
    "    gene_p_val_list.append(gene_p_val)\n",
    "\n",
    "# Get the tissues that have parcel and community assignment in Schaffer\n",
    "valid_tissue = list(donor_gene_parcel_location[donor_gene_parcel_location['community'] != 'NotAssigned']['structure_id'].astype(str))\n",
    "valid_tissue_info = donor_gene_parcel_location[donor_gene_parcel_location['community'] != 'NotAssigned'][['parcel','community']]\n",
    "valid_tissue.insert(0,'gene_symbol')\n",
    "gene_overlap_parcel_exp = gene_overlap_exp.loc[:, valid_tissue].T\n",
    "gene_overlap_parcel_exp.columns = gene_overlap_parcel_exp.iloc[0].values\n",
    "gene_overlap_parcel_exp = gene_overlap_parcel_exp.drop(['gene_symbol'],axis=0)\n",
    "valid_tissue_info.index = gene_overlap_parcel_exp.index\n",
    "\n",
    "# Concatenate two DFs into one\n",
    "gene_overlap_parcel_exp = pd.concat([gene_overlap_parcel_exp,valid_tissue_info],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [],
   "source": [
    "parcel_gene_weighted = []\n",
    "gene_p_val_list_norm = preprocessing.normalize(-np.log10(gene_p_val_list).reshape(1,-1),norm='max')\n",
    "gene_p_val_list_log = -np.log10(gene_p_val_list)\n",
    "\n",
    "for community in gene_overlap_parcel_exp['community'].unique():\n",
    "    for idx, row in gene_overlap_parcel_exp.iterrows():\n",
    "        if row['community'] == community:\n",
    "            weighted_exp = np.dot(row[:-2],gene_p_val_list_norm[0])\n",
    "            #print 'parcel %s in %s with weighted gene expression level at:%s' % (row['parcel'], community, weighted_exp)\n",
    "            #print 'parcel %s in %s with weighted gene expression level at:%s' % (row['parcel'], community, weighted_exp)\n",
    "            parcel_gene_weighted.append({'parcel': row['parcel'], \n",
    "                                        'community': community,\n",
    "                                        'weighted_exp': weighted_exp})\n",
    "parcel_gene_weighted = pd.DataFrame(parcel_gene_weighted)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>parcel</th>\n",
       "      <th>weighted_exp</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>community</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Cont</th>\n",
       "      <td>34.888889</td>\n",
       "      <td>20.880319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Default</th>\n",
       "      <td>43.560976</td>\n",
       "      <td>20.209459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DorsAttn</th>\n",
       "      <td>18.476190</td>\n",
       "      <td>21.243581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Limbic</th>\n",
       "      <td>31.857143</td>\n",
       "      <td>19.940234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SalVentAttn</th>\n",
       "      <td>26.758621</td>\n",
       "      <td>20.335134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SomMot</th>\n",
       "      <td>12.588235</td>\n",
       "      <td>20.577618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vis</th>\n",
       "      <td>3.066667</td>\n",
       "      <td>19.227033</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                parcel  weighted_exp\n",
       "community                           \n",
       "Cont         34.888889     20.880319\n",
       "Default      43.560976     20.209459\n",
       "DorsAttn     18.476190     21.243581\n",
       "Limbic       31.857143     19.940234\n",
       "SalVentAttn  26.758621     20.335134\n",
       "SomMot       12.588235     20.577618\n",
       "Vis           3.066667     19.227033"
      ]
     },
     "execution_count": 239,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parcel_gene_weighted.groupby(by='community').mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "myenv2",
   "language": "python",
   "name": "myenv2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
